src/utils/registry.ts

```diff
@@ -1,10 +1,19 @@
-import * as fs from 'fs';
+import { promises as fsPromises, existsSync } from 'fs';
```

Comment:
The changes in this file are to make these operations async so we can concurrently read/write these manifest files.